rct229/utils/std_comparisons.py

```diff
@@ -29,3 +34,77 @@ def std_equal(
         True iff the val is within percent_tolerance of std_val
     """
     return abs(std_val - val) <= (percent_tolerance / 100) * abs(std_val)
+
+
+def std_equal_with_precision(
+    val: Quantity, std_val: Quantity, precision: Quantity
+) -> bool:
+    """Determines whether the model value and standard value are equal with the specified precision.
+
+    Parameters
+    ----------
+    val: Quantity
+        value extracted from model
+    std_val : Quantity
+        standard value from code
+    precision: Quantity
+        number of decimal places or significant value to round to, and intended units of the comparison
+
+    Returns
+    -------
+    bool
+        True if the modeled value is equal to the standard value within the specified precision
+    """
+
+    # Determine if the values are pint Quantities and handle accordingly
+    if (
+        isinstance(val, Quantity)
+        and isinstance(std_val, Quantity)
+        and isinstance(precision, Quantity)
+    ):
+        units = precision.units
+        val = val.to(units)
+        std_val = std_val.to(units)
+        val_magnitude = Decimal(str(val.magnitude))
+        std_val_magnitude = Decimal(str(std_val.magnitude))
+        precision_magnitude = Decimal(str(precision.magnitude))
+    else:
+        val_magnitude = Decimal(str(val))
+        std_val_magnitude = Decimal(str(std_val))
+        precision_magnitude = Decimal(str(precision))
+
+    # Determine rounding precision based on whether precision is a whole number or a decimal
+    if precision_magnitude.as_tuple().exponent < 0:
+        # Decimal places (e.g., 0.01)
+        precision_decimal_places = abs(precision_magnitude.as_tuple().exponent)
+        rounding_precision = "1E-" + str(precision_decimal_places)
```

Comment:
Stylist comment, I slightly prefer `f` than concatenation for readability (a bit subjective). `f"1E-{precision_deciaml_places}"`